addons/api/addon/handlers/client-daemon-handler.js

```diff
@@ -111,17 +110,20 @@ export default class ClientDaemonHandler {
         );
 
         // Return the raw data if we don't push to the store.
-        let records = normalizedPayload.data;
+        let records = normalizedPayload.data.map((record) => ({
+          ...record.attributes,
+          id: record.id,
+        }));
         if (pushToStore) {
           records = store.push(normalizedPayload);
         }
 
-        // Use a proxy array to store the metadata so the store's `query`
-        // array requirement is satisfied.
-        return ArrayProxy.create({
-          content: records,
-          meta: { totalItems: results?.length ?? 0 },
-        });
+        // Set a meta property on the array to store the total items of the results.
+        // This isn't conventional but is better than returning an ArrayProxy
+        // or EmberArray since the ember store query method asserts it has to be an array
+        // so we can't just return an object.
+        records.meta = { totalItems: results?.length ?? 0 };
+        return records;
```

Comment:
I changed the return type to not use an array proxy now since there's not much of a benefit to it versus just attaching it to an array directly besides the fact it can be a gotcha @lisbet-alvarez

Reply:
So what exactly will the response look like now?

Reply:
Should be used the same, it's just a normal array instead of a proxy array now. The data will look the same